### PR TITLE
Limit suspend voting to suspendee's force

### PR DESCRIPTION
**Pull request by XVhc6A**
Suspend was implemented but allowed people outside of the suspendee's team to grief the vote by just randomly voting.  This limits the voter pool to those on the suspendee's team.

To enable this, I had to add information about the suspender to the global table, and to do this I wrapped most of the important suspend-related information into a class 'SuspendTargetInfo'.  There are parts of the global suspend table that could likely be included, but I would like to keep this simple if possible.

### Tested Changes:
- [ ] I've tested the changes locally or with people.
- [x] I've not tested the changes.

### CODE REVIEW

File: commands/suspend.lua [hunk 1]
```diff
@@ -85,20 +86,18 @@ local suspend_token = Token.register(
...
 				result = math.floor( 100*result / total_votes )
 				if result >= 75 and total_votes > 1 then
-					game.print(global.suspend_target .." suspended... (" .. result .. "%)")
-					Server.to_banned_embed(table.concat {global.suspend_target .. ' was suspended ( ' .. result .. ' %)' .. ', vote started by ' .. global.suspend_starter})
-					global.suspended_players[global.suspend_target] = game.ticks_played
-					local playerSuspended = game.get_player(global.suspend_target)
-					global.suspend_target = nil
-					global.suspend_starter = nil
-					global.suspend_voting = {}
+					game.print(global.suspend_target_info.suspendee_player_name .." suspended... (" .. result .. "%)")
```
Comment (clifffrey) [inline commands/suspend.lua:96]: Perhaps make a local variable like `local info = global.suspend_target_info` just to make the code easier to read

File: commands/suspend.lua [hunk 2]
```diff
@@ -107,15 +106,13 @@ local suspend_token = Token.register(
 			end
 			if total_votes == 1 and result == 100 then
 				game.print("Vote to suspend "..global.suspend_target.." has failed because only 1 player voted, need at least 2 votes")
-				Server.to_banned_embed(table.concat {global.suspend_target .. ' was not suspended and vote failed, only 1 player voted, need at least 2 votes, vote started by ' .. global.suspend_starter})
+				Server.to_banned_embed(table.concat {global.suspend_target .. ' was not suspended and vote failed, only 1 player voted, need at least 2 votes, vote started by ' .. global.suspend_target_info.suspender_player_name})
```
Comment (clifffrey) [inline commands/suspend.lua:109]: this still has a "global.suspend_target" that needs to be replaced with a bigger thing I think

File: commands/suspend.lua [hunk 3]
```diff
@@ -107,15 +106,13 @@ local suspend_token = Token.register(
...
 			else
 				game.print("Vote to suspend "..global.suspend_target.." has failed (" .. result .. "%)")
-				Server.to_banned_embed(table.concat {global.suspend_target .. ' was not suspended and vote failed ( ' .. result .. ' %)' .. ', vote started by ' .. global.suspend_starter})
+				Server.to_banned_embed(table.concat {global.suspend_target .. ' was not suspended and vote failed ( ' .. result .. ' %)' .. ', vote started by ' .. global.suspend_target_info.suspender_player_name})
```
Comment (clifffrey) [inline commands/suspend.lua:112]: ditto

**Review by clifffrey — COMMENTED** [review]
Looks good!

In the future, if you are going to make large whitespace changes, please put them in their own commits, to make reviewing easier.

I'll approve after we do some testing

**Comment by developer-8** [comment]
I tested locally on 2 factorio instances and suspend is broken. Recommend you do test locally too.
`/temp/currently-playing/utils/event_core.lua:46: [ERROR] ...py)/temp/currently-playing/maps/biter_battles_v2/gui.lua:623: attempt to concatenate field 'suspend_target' (a nil value) in ...py)/temp/currently-playing/maps/biter_battles_v2/gui.lua:542`

**Comment by XVhc6A** [comment]
> Looks good!
> 
> In the future, if you are going to make large whitespace changes, please put them in their own commits, to make reviewing easier.
> 
> I'll approve after we do some testing

that's my bad, I accidentally formatted it and didn't realize it.  should pay better attention.

**Comment by XVhc6A** [comment]
This is good to go whenever @clifffrey 
I'm pretty busy the next few days, don't have time to squash and make nice right now, but if you want to speed things idc if you do it yourself.

**Review by clifffrey — APPROVED** [review]

**Pull request merged**